src/ctapipe/calib/camera/extractor.py

```diff
@@ -0,0 +1,261 @@
+
+        # Check if the length of the dl1 table is greater or equal than the size of the chunk.
+        if len(dl1_table[col_name]) < self.chunk_size:
+            raise ValueError(
+                f"The length of the DL1 table ({len(dl1_table[col_name])}) must be greater or equal than the size of the chunk ({self.chunk_size})."
+            )
+        # If no chunk_shift is provided, the chunk_shift is set to self.chunk_size
+        # meaning that the extraction chunks are not overlapping.
+        if chunk_shift is None:
+            chunk_shift = self.chunk_size
+
+        # Function to split table data into appropriated chunks
+        def _get_chunks(dl1_table_data):
+            return [
+                (
+                    dl1_table_data[i : i + self.chunk_size]
+                    if i + self.chunk_size <= len(dl1_table_data)
+                    else dl1_table_data[
+                        len(dl1_table_data) - self.chunk_size : len(dl1_table_data)
+                    ]
+                )
+                for i in range(0, len(dl1_table_data), chunk_shift)
+            ]
+
+        # Get the chunks for the timestamps and selected column name
+        time_chunks = _get_chunks(dl1_table["time_mono"])
+        image_chunks = _get_chunks(dl1_table[col_name].data)
+
+        # Calculate the statistics from a chunk of images
+        stats_list = []
+        for images, times in zip(image_chunks, time_chunks):
+            stats_list.append(
+                self._extract(images, times, masked_pixels_of_sample, outlier_check)
+            )
+        return stats_list
+
+    @abstractmethod
+    def _extract(
+        self, images, times, masked_pixels_of_sample, outlier_check
+    ) -> StatisticsContainer:
+        pass
+
+
+class PlainExtractor(StatisticsExtractor):
+    """
+    Extract the statistics from a chunk of images
+    using numpy and scipy functions
+    """
+
+    def _extract(
+        self, images, times, masked_pixels_of_sample, outlier_check
+    ) -> StatisticsContainer:
+        # ensure numpy array
+        masked_images = np.ma.array(images, mask=masked_pixels_of_sample)
+
+        # median over the sample per pixel
+        pixel_median = np.ma.median(masked_images, axis=0)
+
+        # mean over the sample per pixel
+        pixel_mean = np.ma.mean(masked_images, axis=0)
+
+        # std over the sample per pixel
+        pixel_std = np.ma.std(masked_images, axis=0)
+
+        # outliers from median
+        image_median_outliers = np.logical_or(
+            pixel_median < self.image_median_cut_outliers[0],
+            pixel_median > self.image_median_cut_outliers[1],
+        )
+
+        return StatisticsContainer(
+            extraction_start=times[0],
+            extraction_stop=times[-1],
+            mean=pixel_mean.filled(np.nan),
+            median=pixel_median.filled(np.nan),
+            median_outliers=image_median_outliers.filled(True),
+            std=pixel_std.filled(np.nan),
+        )
```

Comment:
The `PlainExtractor` is used for the peak arrival time correction in the [lstchain code](https://github.com/cta-observatory/cta-lstchain/blob/main/lstchain/calib/camera/flatfield.py#L269-L309). The outliers are only identified based on the medians and not on the stds. Therefore `std_outliers` is empty. We should still fill the `StatisticsContainer` by setting `std_outliers` to `False`, i.e. adding:
`std_outliers=np.full(np.shape(image_median_outliers), False),`